### DISCUSSION TITLE
integratedTerminal does not reopen if closed

### DISCUSSION THREAD

**Discussion by tamighi** [mfussenegger/nvim-dap#1078] [Q&A]
Hello,

When using dap continue the first time with lldb, an integrated terminal opens showing the program output. However, if I close it and relaunch a debug session, it does not open again.

I did not found any api command to open it, nor options to not open it at all (which I don't want, but it would have been nice to have more control over it). I didn't found any related discussion, does someone knows this issue and a fix for it ? I saw that dap ui handles it well but I don't like it's UI and I have trouble understanding it's code.

Thank you in advance!

Here's my config:

local dap = require('dap')
local dap_text = require("nvim-dap-virtual-text")

dap.adapters.codelldb = {
  id = "codelldb",
  type = 'server',
  port = '${port}',
  executable = {
    command = vim.fn.exepath('codelldb'),
    args = { '--port', '${port}' },
  },
}

dap.configurations.cpp = {
  {
    name = "Debug using codelldb",
    type = "codelldb",
    request = "launch",
    program = function()
      return vim.fn.input("Path to executable: ", vim.fn.getcwd() .. "/", "file")
    end,
    cwd = "${workspaceFolder}",
    stopOnEntry = false,
    console = "integratedTerminal",
    args = {},
  },
}

dap.defaults.fallback.terminal_win_cmd = '50vsplit new'

dap_text.setup();

vim.keymap.set("n", "<leader>db", dap.toggle_breakpoint);
vim.keymap.set("n", "<leader>dr", dap.continue);
vim.keymap.set("n", "<leader>dp", dap.pause);
vim.keymap.set("n", "<leader>do", dap.repl.open);
vim.keymap.set("n", "<leader>dd", dap.clear_breakpoints);
vim.keymap.set("n", "<leader>dt", dap.terminate);

vim.keymap.set("n", "<f1>", dap.step_into);
vim.keymap.set("n", "<f2>", dap.step_over);
vim.keymap.set("n", "<f3>", dap.step_out);

**Comment by mfussenegger** [comment]
The integrated terminal uses a regular buffer/window. You can get it back by switching buffers. E.g. via `:bn`
That it doesn't re-open is intentional.

The `terminal_win_cmd` that you configure to a string could also be a function if you want more control over it.

**Reply by tamighi** [reply]
Is the function only called once, and I would get to find my buffer back again if I want the terminal back anyway ? Like with an auto command ?

**Reply by asmodeus812** [reply]
@mfussenegger  i am also facing the same issue even with a function configured, it seems the function is called only once for some reason, if i kill the buffer however when the session is over, on starting the session again the buffer is somehow re-created, without calling my callback again. But is not shown anywhere, just stays hidden.  There should be more control in the config over when and how the terminal buf is created and destroyed imo. You can even replay this with the defaults, if you bd! the terminal buffer it would re-create it but would not trigger the win_cmd command to open it in a split.
